docs/partitioning.md

```diff
@@ -1,52 +1,150 @@
+
+For quick reference, you can cluster your table using a query like this:
+
+```
+#standardSql
+CREATE TABLE `clustered_table`
+PARTITION BY fake_date
+CLUSTER BY reference_name, start_position, end_position AS (
+  SELECT *, DATE('1980-01-01') fake_date
+  FROM `original_table`
+)
+```
+
+Clustering can be very effective in reducing the cost of queries, however,
+it has a few limitations:
+ * Clustering does not offer any *guarantees* on the cost (it's a best effort
+ reduction).
+ * It needs a one time clustering step which can take a few hours for large
+ datasets.
+ * If you append data to an existing clustered table it will become partially
+ sorted. So you need to regularly re-clustering your table.
+
+## Solution 2: Partitioning output table
+
+The second solution for reducing the cost of queries is to use Variant
+Transforms' native partitioning. Variant transforms is able to split the output
+table into several smaller tables, each containing variants of a specific region of a
+genome. For example, you can have one output table per chromosome, in that
+case the above query can be written as:
+
+```
+SELECT * from `1000genomes_chr17`
+WHERE start_position >= 41197694
+  AND end_position <= 41276113
+```
+
+Note that condition on the `reference_name` is removed since we know this table,
+as its name suggests, only contains the variants of 17th chromosome.
+
+By splitting output tables based on the `reference_name` you are
+guaranteed that per-chromosome queries will only process variants of the
+chromosome under study. Also, appending new rows to existing tables does not
+impact this guarantee on the cost, unlike the BigQuery clustering solution.
+
+This solution has some limitations comparing to clustering. For example, you
+will be charged for processing of a whole chromosome's table even if only a
+small region is being processed. As an example, the previous query, will cost
+152 GB or $0.74. This is significantly less than the original cost without
+partitioning but it's more than clustering cost.
+
+You could define your partitions to be more fine grained and have multiple
+tables per chromosome. However, you need to anticipate how your future queries
+are going to be in order to optimize your output partitions. Since in many
+use cases it not obvious to anticipate future queries, we offer the
+third solution as the most practical and cost effective solution.
+
+## Solution 3: Hybrid Solution
+
+This solution combines two previous solutions to offer the benefits of both.
+Using the Variant Transforms' native partitioning output table will be split
+into several smaller tables (perhaps one table per chromosome) and then each
+table will be clustered based on the `start_position, end_position` columns to
+further optimize them for running queries.
+
+Using this technique the *BRCA1 query* will cost 226 MB or $0.0011 which is more
+than half of the clustering cost. In our experimental studies, we found out the
```

Comment:
consider rephrasing the last sentence...e.g.
`In our experiments, we found that the most effective way to reduce query cost, especially for point lookup queries, is this hybrid solution.`

Reply:
done.